src/main/java/mate/academy/internet/shop/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,187 @@
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                user.setId(resultSet.getLong(1));
+            }
+            LOGGER.info("User with id=" + user.getId() + " was created");
+            addRolesForUser(user);
+            return user;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Creation of user is failed", e);
+        }
+    }
+
+    @Override
+    public Optional<User> get(Long id) {
+        String query = "SELECT * from internet_shop.users where user_id=?;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(getCopyOfUser(resultSet));
+            }
+            LOGGER.info("User with id=" + id + " was received");
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Getting of user with id="
+                    + id + " is failed", e);
+        }
+    }
+
+    @Override
+    public List<User> getAll() {
+        List<User> users = new ArrayList<>();
+        String query = "SELECT * FROM internet_shop.users;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                users.add(getCopyOfUser(resultSet));
+            }
+            LOGGER.info("List of users was received");
+            return users;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Getting of list of users is failed", e);
+        }
+    }
+
+    @Override
+    public User update(User user) {
+        String query = "UPDATE internet_shop.users "
+                + "SET name=?, login=?, password=? WHERE user_id=?;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setString(1, user.getName());
+            statement.setString(2, user.getLogin());
+            statement.setString(3, user.getPassword());
+            statement.setLong(4, user.getId());
+            statement.executeUpdate();
+            deleteRolesOfUser(user);
+            addRolesForUser(user);
+            LOGGER.info("User with id=" + user.getId() + " was updated");
+            return user;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Update of user with id="
+                    + user.getId() + " is failed", e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "DELETE FROM internet_shop.users WHERE user_id=?;";
```

Comment:
same question about this situation. You are trying to delete user with references in others tables to this UserId.What will happen?

Reply:
the same